components/learn/modules/ROOT/pages/unit-testing.adoc

```diff
@@ -0,0 +1,174 @@
+
+When it comes to smart contract development though, practice has shown that https://en.wikipedia.org/wiki/Unit_testing[_contract unit testing_] is exceptionally worthwhile. These tests are simple to write and and quick to run, and let you add features and fix bugs in your code with confidence.
+
+Smart contract unit testing consists of multiple small, focused tests, which each check a small part of your contract for correctness. They can often be expressed in single sentences that make up a specification, such as 'the admin is able to pause the contract', 'transferring tokens emits an event' or 'non-admins cannot mint new tokens'.
+
+[[setting-up-a-testing-environment]]
+== Setting up a Testing Environment
+
+You may be wondering _how_ we're going to run these tests, since smart contracts are executed inside a blockchain. Using the actual Ethereum network would be very expensive, and while testnets are free, they are also slow (with blocktimes between 5 and 15 seconds). If we intend to run hundreds of tests whenever we make a change to our code, we need something better.
+
+What we will use is called a _local blockchain_: a slimmed down version of the real thing, disconnected from the Internet, running on your machine. This will simplify things quite a bit: you won't need to go get Ether, and you won't have to wait for blocks to be mined.
+
+To aid us in this we'll use the https://github.com/OpenZeppelin/openzeppelin-test-environment#readme[*OpenZeppelin Test Environment*], a JavaScript library that will take care of setting up our local blockchain.
+
+TIP: If you've read the xref:interact.adoc['Interacting with Your Contracts' guide], you will already be familiar with https://github.com/trufflesuite/ganache-cli/[Ganache]. Test Environment works in a similar way, but is much easier to use.
+
+To install the OpenZeppelin Test Environment, run:
+
+```bash
+$ npm install --save-dev @openzeppelin/test-enviroment
+```
+
+Once you `require` the library from your JavaScript code, it will automatically run a local testing blockchain for you. It also exports variables and functions related to this environment, such as a list of `accounts` that have been pre-funded with Ether, and convenient ways to load your contracts from their compiled artifacts.
+
+```javascript
+const { accounts, contract } = require('@openzeppelin/test-environment#readme');
+
+// Use the different accounts, which are unlocked and funded with Ether
+const [ admin, deployer, user ] = accounts;
+
+// Create a contract object from a compilation artifact
+const MyContract = contract.fromArtifact('MyContract');
+```
+
+For detailed information on these exported values and their usage, refer to Test Environment's https://github.com/OpenZeppelin/openzeppelin-test-environment/blob/master/docs/modules/ROOT/pages/api.adoc[API reference].
```

Comment:
Should we add a _Tip_ block explaining that the default account is not listed? I find it an interesting design decision, and presenting it in an aside block allows the reader to just skip through it. It could also lead to confusion for a user that goes a bit in depth, so documenting it up front may help avoid this.